### PR TITLE
save units in attrs of H5

### PR DESCRIPTION
**Pull request by PythonFZ**
- fix compliancy with H5MD by using `attrs`
- add MDAnalysis test to read file (see https://github.com/MDAnalysis/mdanalysis/issues/4598 for why the test is so minimal)
- deprecate `initialize_database_groups`
- now downstream issues with IPSuite https://github.com/zincware/IPSuite/pull/300

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ pre-commit = "^2.20.0"
 coverage = "^7.1.0"
 pytest-profiling = "^1.7.0"
 pytest-benchmark = "^4.0.0"
+mdanalysis = "^2"
```
Comment (M-R-Schaefer) [inline pyproject.toml:25]: should this be an optional dependency?

File: znh5md/format/__init__.py [hunk 2]
```diff
@@ -67,6 +67,25 @@ def decode_boundary(value) -> np.ndarray:
...
+UNITS_PER_GRP = {
+    GRP.position: StepTimeChunkUnits(value="Angstrom", time="fs"),
+    GRP.velocity: StepTimeChunkUnits(value="Angstrom/fs", time="fs"),
+    GRP.forces: StepTimeChunkUnits(value="eV/Angstrom", time="fs"),
+    GRP.energy: StepTimeChunkUnits(value="eV", time="fs"),
+    GRP.stress: StepTimeChunkUnits(value="GPa", time="fs"),
+    GRP.momentum: StepTimeChunkUnits(value="eV/fs", time="fs"),
+    GRP.edges: StepTimeChunkUnits(value="Angstrom", time="fs"),
+    GRP.boundary: StepTimeChunkUnits(value="Angstrom", time="fs"),
+}
```
Comment (M-R-Schaefer) [inline znh5md/format/__init__.py:86]: ah so these are like metadata in the H5 file?

**Review by M-R-Schaefer — APPROVED** [review]
looks good. But this will break existing IPS H5 files, right? Is there some sort of backward compatibility functionality?
If not it?s fine too, I just need to breace myself :joy:

File: pyproject.toml (hunk 1 above)
Comment (PythonFZ) [reply pyproject.toml:25]: Its is in the `[tool.poetry.group.dev.dependencies]`

File: znh5md/format/__init__.py (hunk 2 above)
Comment (PythonFZ) [reply znh5md/format/__init__.py:86]: Yes, these are the ASE units. This is a bit inflexible right now, such that if you don't go through ASE there is no nice way of defining units.

**Comment by PythonFZ** [comment]
> looks good. But this will break existing IPS H5 files, right? Is there some sort of backward compatibility functionality? If not it?s fine too, I just need to breace myself 😂

Reading should be fully backwards compatible

**Pull request merged**